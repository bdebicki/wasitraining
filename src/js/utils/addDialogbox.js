'use strict';
import { DIALOGBOX, LINK } from '../enums/elementHandlers';
import { setIcon } from './setIcon';
import { icons } from './iconsLibrary';

export function addDialogbox({id, classNames, closeAction = toggleDialogbox, closeTitle = 'hide details', title, content} = {}) {
	let div = document.createElement('div');
	let header = document.createElement('header');
	let h3 = document.createElement('h3');
	let closeBtn = document.createElement('a');

	const dialogboxId = id;
	const dialogboxClassNames = [DIALOGBOX.BASIC_CLASS];

	classNames.map((item) => {
		dialogboxClassNames.push(item);
	});

	div.id = dialogboxId;
	div.classList.add(...dialogboxClassNames);
	closeBtn.classList.add(DIALOGBOX.CLOSE_CLASS);
	closeBtn.appendChild(setIcon(icons.close(), `${LINK.ICON_CLASS}`));
	closeBtn.setAttribute('href', dialogboxId);
	closeBtn.addEventListener('click', closeAction, null);
	closeBtn.title = closeTitle;
	header.classList.add(DIALOGBOX.HEADER_CLASS);
	h3.classList.add(DIALOGBOX.HEADLINE_CLASS);
	h3.textContent = title;

	header.appendChild(h3);
	header.appendChild(closeBtn);
	div.appendChild(header);
	div.appendChild(content);

	return div;
}

export function toggleDialogbox(e) {
	e.preventDefault();
	toggleDialogboxAction(`.${DIALOGBOX.VISIBLE_CLASS}`);
}

export function toggleDialogboxAction(target) {
	document.querySelector(target).classList.toggle(DIALOGBOX.VISIBLE_CLASS);
}