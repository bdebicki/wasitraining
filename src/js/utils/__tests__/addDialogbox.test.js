import using from '../../../../testUtils/using';
import { addDialogbox } from '../addDialogbox';

function mockDialogboxContent() {
	const dialogboxContent = document.createElement('div');
	dialogboxContent.textContent = 'Simple dialogbox content';

	return dialogboxContent;
}

const dialogboxCases = [
	{
		description: 'render dialogbox with default and required propertis settings',
		settings: {
			id: 'lineupDialogbox',
			title: 'Lineup 2018',
			content: mockDialogboxContent(),
		},
	}, {
		description: 'render dialogbox with overwrite propertis with default values (stretched, closeTitle)',
		settings: {
			id: 'lineupDialogbox',
			title: 'Lineup 2018',
			content: mockDialogboxContent(),
			stretched: true,
			closeTitle: 'close dialogbox',
		},
	}, {
		description: 'render dialogbox with additional properties (classNames, adataAttr)',
		settings: {
			id: 'lineupDialogbox',
			title: 'Lineup 2018',
			content: mockDialogboxContent(),
			classNames: ['dialogbox-visible', 'lineup--edition2018'],
			dataAttr: [['year', '2018']],
		},
	},
];

describe('render dialogbox element', () => {
	using(dialogboxCases).describe('', ({ description, settings }) => {
		it(description, () => {
			const dialogboxEl = addDialogbox(settings);

			expect(dialogboxEl).toMatchSnapshot();
		});
	});
});
