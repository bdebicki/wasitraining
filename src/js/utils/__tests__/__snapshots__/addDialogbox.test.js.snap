// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render dialogbox element  render dialogbox with additional properties (classNames, adataAttr) 1`] = `
<div
  class="dialogbox-visible lineup--edition2018"
  data-year="2018"
  id="lineupDialogbox"
>
  <header
    class="dialogbox__header"
  >
    <h3
      class="dialogbox__headline"
    >
      Lineup 2018
    </h3>
    <a
      class="dialogbox__close"
      href="lineupDialogbox"
      title="hide details"
    >
      <svg
        class="icon--close link__icon"
        height="10px"
        viewBox="0 0 10 10"
        width="10px"
      >
        <rect
          class="icon--close__line"
          height="2"
          width="12"
          x="-1"
          y="4"
        />
        <rect
          class="icon--close__line"
          height="12"
          width="2"
          x="4"
          y="-1"
        />
      </svg>
    </a>
  </header>
  <div>
    Simple dialogbox content
  </div>
</div>
`;

exports[`render dialogbox element  render dialogbox with default and required propertis settings 1`] = `
<div
  class="dialogbox"
  id="lineupDialogbox"
>
  <header
    class="dialogbox__header"
  >
    <h3
      class="dialogbox__headline"
    >
      Lineup 2018
    </h3>
    <a
      class="dialogbox__close"
      href="lineupDialogbox"
      title="hide details"
    >
      <svg
        class="icon--close link__icon"
        height="10px"
        viewBox="0 0 10 10"
        width="10px"
      >
        <rect
          class="icon--close__line"
          height="2"
          width="12"
          x="-1"
          y="4"
        />
        <rect
          class="icon--close__line"
          height="12"
          width="2"
          x="4"
          y="-1"
        />
      </svg>
    </a>
  </header>
  <div>
    Simple dialogbox content
  </div>
</div>
`;

exports[`render dialogbox element  render dialogbox with overwrite propertis with default values (stretched, closeTitle) 1`] = `
<div
  class="dialogbox dialogbox--stretched"
  id="lineupDialogbox"
>
  <header
    class="dialogbox__header"
  >
    <h3
      class="dialogbox__headline"
    >
      Lineup 2018
    </h3>
    <a
      class="dialogbox__close"
      href="lineupDialogbox"
      title="close dialogbox"
    >
      <svg
        class="icon--close link__icon"
        height="10px"
        viewBox="0 0 10 10"
        width="10px"
      >
        <rect
          class="icon--close__line"
          height="2"
          width="12"
          x="-1"
          y="4"
        />
        <rect
          class="icon--close__line"
          height="12"
          width="2"
          x="4"
          y="-1"
        />
      </svg>
    </a>
  </header>
  <div>
    Simple dialogbox content
  </div>
</div>
`;
